src/datepicker/datepicker-input.ts

```diff
@@ -306,8 +306,10 @@ export class NgbInputDatepicker implements OnChanges,
   }
 
   private _applyPopupStyling(nativeElement: any) {
+    this._renderer.removeClass(nativeElement, 'd-inline-block');
```

Comment:
Hmmm, do we need to do this since the datepicker widget has `d-inline-block` on host?

Reply:
Yup. Cause some themes will not display this element. For example, they make it not visible by scaling it to 0 and changing opacity to 0, so display doesn't matter at all.

Reply:
We need to make it truly not visible, so it will have animation once made visible by class "show". It won't be a case with opacity 0, but not all themes use it.

Reply:
Hi
There is no more the d-inline-block class on the datepicker element, I removed it in pull request 
#1614
This pull request will solve #2088.
BR

Reply:
Ok, so I will remove this line in next commit